prime-router/src/main/kotlin/azure/BlobAccess.kt

```diff
@@ -234,6 +242,75 @@ class BlobAccess() : Logging {
             val previousBlobItemVersions: List<BlobItem>?,
         )
 
+        enum class SchemaType {
+            FHIR_FHIR,
+            FHIR_HL7,
+        }
+
+        /**
+         * Downloads the content of each blob in the [directory] with [blobContainerInfo] takes the input file, runs
+         * the transform on it, and compares it to the output validating that they are the same.
+         */
+        fun validateSchemas(
+            directory: String,
+            blobContainerInfo: BlobContainerMetadata = defaultBlobMetadata,
+            schemaType: SchemaType,
+        ): Boolean {
+            val sourceContainer = getBlobContainer(blobContainerInfo)
+            val blobs =
+                listBlobs(directory, blobContainerInfo, false)
+            val transforms = blobs.filter { it.currentBlobItem.name.endsWith(".yml") }
+            transforms.forEach { currentTransform ->
+                val currentTransformName = currentTransform.currentBlobItem.name
+
+                val transformBlobClient = sourceContainer.getBlobClient(currentTransformName)
+                val transform = transformBlobClient.downloadContent()
+
+                val regex = Regex("(.*/).*")
+                val path = regex.find(currentTransformName)!!.groups[1]!!.value
+                val inputBlobClient = sourceContainer.getBlobClient(path + "input.fhir")
+                val input = inputBlobClient.downloadContent()
+                val inputBundle = FhirTranscoder.decode(input.toString())
+
+                if (schemaType == SchemaType.FHIR_FHIR) {
+                    val configSchema = ConfigSchemaReader.readOneYamlSchema(
+                        transform.toStream(),
+                        FhirTransformSchema::class.java
+                    )
+                    val fhirTransform = if (configSchema is FhirTransformSchema) {
+                        FhirTransformer(configSchema).transform(inputBundle)
+                    } else {
+                        throw InternalServerErrorException("Transform schema $currentTransformName not a FhirTransform")
+                    }
+                    val output = sourceContainer.getBlobClient(path + "output.fhir").downloadContent()
+                    // Unfortunately need to do this to get it to be the same format.
+                    val finalOutput = FhirTranscoder.encode(FhirTranscoder.decode(output.toString()))
+                    val finalTransform = FhirTranscoder.encode(fhirTransform)
+                    // .equals is required here == does not compare it properly, despite what IntelliJ says
+                    if (!finalTransform.equals(finalOutput)) {
+                        logger.error("Validation failed for transform $currentTransformName")
+                        return false
+                    }
+                } else {
+                    val converterSchema = ConfigSchemaReader.readOneYamlSchema(transform.toStream())
```

Comment:
The way `ConfigSchemaReader` is designed to be used is via `fromFile`. `fromFile` will merge schemas that are referenced into a final schema, and since we are calling `readOneYamlSchema`, we end up skipping that functionality... I think we will need to figure out how to resolve this issue, unless the design notes referenced schemas will not be supported?

Reply:
I do not understand this comment, let's discuss.